aten/src/ATen/ParallelNative.cpp

```diff
@@ -101,30 +150,46 @@ int get_num_threads() {
     TORCH_INTERNAL_ASSERT(nthreads == CONSUMED);
     return internal::_get_intraop_pool().size() + 1;
   }
+#else
+  caffe2::ThreadPool* pool = caffe2::mobile_threadpool();
+  // caffe2::ThreadPool::getNumThreads() counts the current thread.
+  return !pool ? 1 /* current thread */ : pool->getNumThreads();
+#endif // C10_MOBILE
 }
 
 int get_thread_num() {
   return thread_num_;
 }
 
 bool in_parallel_region() {
+#ifndef C10_MOBILE
   return in_parallel_region_ || (
     num_intraop_threads.load() == CONSUMED &&
     internal::_get_intraop_pool().inThreadPool()
   );
```

Comment:
we could just unify this as `return in_parallel_region_`

Reply:
I didn't change it because intraop_launch() / intraop_launch_future() don't set in_parallel_region_ - technically those callback functions could call parallel APIs too. We don't implement intraop_launch() for mobile (yet) so it's not an issue for mobile.